### ISSUE TITLE
Validation errors not displayed at document init.

### ISSUE THREAD

**Issue by CubixSystem** [keystonejs/keystone#1437]
Validation error on validate hook doesn't create any error message like this:
![10](https://cloud.githubusercontent.com/assets/7340055/7924691/2e5327d8-08c6-11e5-8a97-8aaa296131ca.PNG)

Example:
```javascript
Player.schema.pre("validate", function (next) {
    next(new Error("Test"));
});
```

In console:
```javascript
Error saving changes to Player 556ce5a7a97fabbc287c6aa8:
[Error: Test]
```

**Comment by grepped** [comment]
Did you find a fix for this issue? Currently experiencing this

**Comment by AlexYegupov** [comment]
Made [quick fix](https://github.com/AlexYegupov/keystone/commit/29334e9599322156f8a7f08a1edf6d85b465b576#diff-1216fa2cb9fb0cbc959b8bdba7272512)  in own fork in v0.3.x branch

It couldn't be considered as correct patch but seems to work for me.
Problems that should be solved:
 - format of err structure should be improved somehow
 - pre-validate functions are called only after all field are validated and in turn (if there are several ones)

**Comment by mxstbr** [comment]
Fixed in #2566!
